File: app/src/main/java/com/rokudo/xpense/data/viewmodels/WalletsViewModel.java
package com.rokudo.xpense.data.viewmodels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

import com.rokudo.xpense.data.repositories.WalletsRepo;
import com.rokudo.xpense.models.Wallet;

import java.util.ArrayList;

public class WalletsViewModel extends AndroidViewModel {
    private final WalletsRepo repo;

    public WalletsViewModel(@NonNull Application application) {
        super(application);
        repo = WalletsRepo.getInstance();
    }

    public MutableLiveData<ArrayList<Wallet>> loadWallets() {
        return repo.getWallets();
    }

    public MutableLiveData<Wallet> loadWallet(String walletId) {
        return repo.loadWallet(walletId);
    }

    public MutableLiveData<String> addWallet(Wallet wallet) {
        return repo.addWallet(wallet);
    }

    public MutableLiveData<Boolean> updateWallet(Wallet wallet) {
        return repo.updateWallet(wallet);
    }

    public MutableLiveData<Boolean> deleteWallet(String walletId) {
        return repo.deleteWallet(walletId);
    }

    public void removeAllData() {
        repo.removeAllWalletsData();
    }
}

File: app/src/main/java/com/rokudo/xpense/data/viewmodels/WalletsViewModel.kt
package com.rokudo.xpense.data.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.MutableLiveData
import com.rokudo.xpense.data.repositories.WalletsRepo
import com.rokudo.xpense.models.Wallet

class WalletsViewModel(application: Application) : AndroidViewModel(application) {
    private val repo = WalletsRepo.instance

    fun loadWallets(): MutableLiveData<ArrayList<Wallet>?> = repo.getWallets()

    fun loadWallet(walletId: String): MutableLiveData<Wallet?> = repo.loadWallet(walletId)

    fun addWallet(wallet: Wallet): MutableLiveData<String> = repo.addWallet(wallet)

    fun updateWallet(wallet: Wallet): MutableLiveData<Boolean> = repo.updateWallet(wallet)

    fun deleteWallet(walletId: String): MutableLiveData<Boolean> = repo.deleteWallet(walletId)

    fun removeAllData() = repo.removeAllWalletsData()
}

